### PR TITLE
GH-512 python: % -> format(); other small refactoring

### PR DESCRIPTION
**Pull request by adrianopol**

### CODE REVIEW

File: scripts/scons_plugin/commands.py [hunk 1]
```diff
@@ -38,7 +40,7 @@ def Doxygen(env, build_dir='', html_dir=None, config='', sources=[], werror=Fals
         dirs += [env.Dir(html_dir).path]
 
     env.Command(target, sources + [config], SCons.Action.CommandAction(
-        '%s scripts/scons_helpers/docfilt.py %s %s %s %s %s %s %s' % (
+        '{} scripts/scons_helpers/docfilt.py {} {} {} {} {} {} {}'.format(
```
Comment (gavv) [inline scripts/scons_plugin/commands.py:43]: Let's use named format args here.

File: scripts/scons_plugin/commands.py [hunk 2]
```diff
@@ -55,7 +57,8 @@ def Sphinx(env, output_type, build_dir, output_dir, source_dir, sources, werror=
     target = os.path.join(build_dir, '.done')
 
     env.Command(target, sources, SCons.Action.CommandAction(
-        '%s scripts/scons_helpers/docfilt.py %s %s %s %s %s %s -j %d -q -b %s -d %s %s %s' % (
+        '{} scripts/scons_helpers/docfilt.py ' +
+            '{} {} {} {} {} {} -j {} -q -b {} -d {} {} {}'.format(
```
Comment (gavv) [inline scripts/scons_plugin/commands.py:61]: Let's use named format args here.

File: scripts/scons_plugin/commands.py [hunk 3]
```diff
@@ -88,7 +91,7 @@ def Ragel(env, source):
     target = os.path.join(str(source.dir), target_name)
 
     env.Command(target, source, SCons.Action.CommandAction(
-        '%s -o "%s" "%s"' % (
+        '{} -o "{}" "{}"'.format(
```
Comment (gavv) [inline scripts/scons_plugin/commands.py:94]: Let's use named format args here.

File: scripts/scons_plugin/commands.py [hunk 4]
```diff
@@ -114,7 +117,7 @@ def GenGetOpt(env, source, ver):
     ]
 
     env.Command(target, source, SCons.Action.CommandAction(
-        '%s -i "%s" -F "%s" --output-dir "%s" --set-version "%s"' % (
+        '{} -i "{}" -F "{}" --output-dir "{}" --set-version "{}"'.format(
```
Comment (gavv) [inline scripts/scons_plugin/commands.py:120]: Let's use named format args here.

File: scripts/scons_plugin/commands.py [hunk 5]
```diff
@@ -205,15 +208,15 @@ def NeedsFixupSharedLibrary(env):
...
+            '$INSTALL_NAME_TOOL -id "{0}" "{0}"'.format(path),
             cmdstr=env.PrettyCommand('FIXUP', path, 'yellow')),
             ]
 
 def ComposeStaticLibraries(env, dst_lib, src_libs):
     dst_lib = env['LIBPREFIX'] + dst_lib + env['LIBSUFFIX']
 
     action = SCons.Action.CommandAction(
-        '%s scripts/scons_helpers/compose-libs.py %s %s AR=%s' % (
+        '{} scripts/scons_helpers/compose-libs.py {} {} AR={}'.format(
```
Comment (gavv) [inline scripts/scons_plugin/commands.py:219]: Let's use named format args here.

File: scripts/scons_plugin/tests.py [hunk 6]
```diff
@@ -18,14 +18,14 @@ def _get_non_test_targets(env):
...
+    return '{} scripts/scons_helpers/run-with-timeout.py {} {}'.format(
         env.GetPythonExecutable(),
         timeout,
         cmd)
 
 def _add_test(env, kind, name, exe, cmd, timeout):
-    varname = '_%s_TARGETS' % kind.upper()
-    testname = '%s/%s' % (kind, name)
+    varname = '_{}_TARGETS'.format(kind.upper())
+    testname = os.path.join(kind, name)
```
Comment (gavv) [inline scripts/scons_plugin/tests.py:28]: It should be `/`, not os.path.join. It's target name that can be passed to scons, like `scons test/roc_core`, not a path.

File: scripts/scons_plugin/thirdparty.py [hunk 7]
```diff
@@ -40,7 +40,7 @@ def _build_thirdparty(env, build_root, toolchain, variant, versions, name, deps,
         os.chdir(project_root)
 
         if env.Execute(
-            '%s scripts/scons_helpers/build-3rdparty.py %s 3rdparty/_distfiles %s %s %s %s %s' % (
+            '{} scripts/scons_helpers/build-3rdparty.py {} 3rdparty/_distfiles {} {} {} {} {}'.format(
```
Comment (gavv) [inline scripts/scons_plugin/thirdparty.py:43]: Let's use named format args here.

File: scripts/scons_plugin/thirdparty.py [hunk 8]
```diff
@@ -79,22 +79,22 @@ def needlib(lib):
...
-            incdir += '/' + s
+            incdir = os.path.join(incdir, s)
 
         if os.path.isdir(env.Dir(incdir).abspath):
             env.Prepend(CPPPATH=[incdir])
 
-    libdir = '%s/%s/lib' % (build_root, versioned_name)
-    rpathdir = '%s/%s/rpath' % (build_root, versioned_name)
+    libdir = os.path.join(build_root, versioned_name, 'lib')
+    rpathdir = os.path.join(build_root, versioned_name, 'rpath')
```
Comment (gavv) [inline scripts/scons_plugin/thirdparty.py:90]: Here and in other places, let's revert os.path.join.

These paths are passed to scons and I think they should use `/`

**Comment by gavv** [comment]
I most cases where `%s` was replaced with `+`, I find the new version is less readable.

For example here:

```python
'-Wl,-soname,libroc%s' % subenvs.public_libs['SHLIBSUFFIX']
```

I immediately see that the resulting string is "libroc<something>", but here:

```python
'-Wl,-soname,libroc' + subenvs.public_libs['SHLIBSUFFIX']
```

I need to read the whole expression to see what resulting string looks like.

Please consider using `format` instead of `+` in such cases.

This looks OK:

```python
'-Wl,-soname,libroc{}'.format(subenvs.public_libs['SHLIBSUFFIX'])
```

**Comment by gavv** [comment]
In many (most?) places where `/` was replaced with os.path.join, it was done incorrectly.

When we pass paths to scons (e.g. to CPPPATH), IIRC we should use `/`.

Anyway, let's avoid this change in this PR, this is out of its scope.

File: 3rdparty/SConscript [hunk 9]
```diff
@@ -344,7 +344,7 @@ elif 'ragel' in system_dependencies:
         ragel = 'ragel'
 
     if not conf.CheckProg(ragel):
-        env.Die("ragel not found in PATH (looked for '%s')" % ragel)
+        env.Die("ragel not found in PATH (looked for '{}')".format(ragel))
```
[the remark this answers is not in the document]
Comment (adrianopol) [reply 3rdparty/SConscript:347]: fixed

File: 3rdparty/SConscript [hunk 10]
```diff
@@ -365,7 +365,7 @@ elif 'gengetopt' in system_dependencies:
         gengetopt = 'gengetopt'
 
     if not conf.CheckProg(gengetopt):
-        env.Die("gengetopt not found in PATH (looked for '%s')" % gengetopt)
+        env.Die("gengetopt not found in PATH (looked for '{}')".format(gengetopt))
```
[the remark this answers is not in the document]
Comment (adrianopol) [reply 3rdparty/SConscript:368]: fixed

File: SConstruct [hunk 11]
```diff
@@ -440,22 +440,22 @@ else:
             meta.compiler = 'cc'
 
 if not meta.compiler:
-    env.Die("can't detect compiler name, please specify '--compiler={name}' manually,"+
-            " should be one of: %s'" % ', '.join(supported_compilers))
+    env.Die("can't detect compiler name, please specify '--compiler={name}' manually, "
+            "should be one of: {}'".format(', '.join(supported_compilers)))
```
[the remark this answers is not in the document]
Comment (adrianopol) [reply SConstruct:444]: fixed

File: SConstruct [hunk 12]
```diff
@@ -514,11 +514,11 @@ if not meta.platform and meta.host == meta.build:
         meta.platform = 'unix'
 
 if not meta.platform:
-    env.Die("can't detect platform name, please specify '--platform={name}' manually,"+
-            " should be one of: %s'" % ', '.join(supported_platforms))
+    env.Die("can't detect platform name, please specify '--platform={name}' manually, "
+            "should be one of: {}'".format(', '.join(supported_platforms)))
```
[the remark this answers is not in the document]
Comment (adrianopol) [reply SConstruct:518]: fixed

File: scripts/scons_plugin/clangdb.py [hunk 13]
```diff
@@ -1,7 +1,7 @@
 import os
 
 def GetClangDbWriter(env, tool, build_dir):
-    return '%s scripts/scons_helpers/clangdb.py "%s" "%s" "%s"' % (
+    return '{} scripts/scons_helpers/clangdb.py "{}" "{}" "{}"'.format(
```
[the remark this answers is not in the document]
Comment (adrianopol) [reply scripts/scons_plugin/clangdb.py:4]: done

File: scripts/scons_plugin/commands.py (hunk 3 above)
Comment (adrianopol) [reply scripts/scons_plugin/commands.py:94]: done

File: scripts/scons_plugin/commands.py (hunk 4 above)
Comment (adrianopol) [reply scripts/scons_plugin/commands.py:120]: done

File: scripts/scons_plugin/commands.py (hunk 5 above)
Comment (adrianopol) [reply scripts/scons_plugin/commands.py:219]: done

File: scripts/scons_plugin/tests.py (hunk 6 above)
Comment (adrianopol) [reply scripts/scons_plugin/tests.py:28]: fixed

File: scripts/scons_plugin/thirdparty.py (hunk 7 above)
Comment (adrianopol) [reply scripts/scons_plugin/thirdparty.py:43]: done

**Comment by gavv** [comment]
LGTM

**Pull request merged**